### PR TITLE
Add flag to create missing directories in output path.

### PR DESCRIPTION
**Pull request by zurk**
Instead of https://github.com/src-d/ml/pull/211

### CODE REVIEW

File: modelforge/model.py [hunk 1]
```diff
@@ -256,14 +256,20 @@ def dump(self) -> str:
         """
         raise NotImplementedError()
 
-    def save(self, output: Union[str, BinaryIO], deps: Iterable=tuple()) -> "Model":
+    def save(self, output: Union[str, BinaryIO], deps: Iterable=tuple(),
+                   create_missing_dirs: bool=True) -> "Model":
```
Comment (vmarkovtsev) [inline modelforge/model.py]: Looks like not aligned to `save(`

File: modelforge/model.py [hunk 2]
```diff
@@ -256,14 +256,20 @@ def dump(self) -> str:
...
         """
         Serializes the model to a file.
 
         :param output: path to the file or a file object.
         :param deps: the list of the dependencies.
+        :param create_missing_dirs: Create missing directories in output path.
```
Comment (vmarkovtsev) [inline modelforge/model.py]: ... if the output is a path
**create** (lower case first letter)

File: modelforge/model.py [hunk 3]
```diff
@@ -256,14 +256,20 @@ def dump(self) -> str:
...
         :return: self
         """
+        if create_missing_dirs:
```
Comment (vmarkovtsev) [inline modelforge/model.py]: You need to check if `output` is a string

File: modelforge/model.py [hunk 4]
```diff
@@ -256,14 +256,21 @@ def dump(self) -> str:
...
-    def save(self, output: Union[str, BinaryIO], deps: Iterable=tuple()) -> "Model":
+    def save(self, output: Union[str, BinaryIO], deps: Iterable=tuple(),
+             create_missing_dirs: bool=True) -> "Model":
         """
         Serializes the model to a file.
 
         :param output: path to the file or a file object.
         :param deps: the list of the dependencies.
+        :param create_missing_dirs: create missing directories in output path if the output is a
+                                    path.
```
Comment (vmarkovtsev) [inline modelforge/model.py]: Docstrings require `\` to break lines

**Comment by vmarkovtsev** [comment]
@zurk pep8 and I will merge this, also the docstring thing

File: modelforge/model.py [hunk 5]
```diff
@@ -338,8 +338,8 @@ def merge_strings(list_of_strings: Union[List[str], Tuple[str]]) -> dict:
     max_len = 0
     lengths = []
     for s in list_of_strings:
-        l = len(s)
-        lengths.append(l)
+        length = len(s)
+        lengths.append(length)
         if l > max_len:
```
Comment (vmarkovtsev) [inline modelforge/model.py]: `l` is used here

File: modelforge/model.py [hunk 6]
```diff
@@ -68,7 +68,7 @@ def load(self, source: Union[str, BinaryIO, "Model"]=None,
                     if self.NAME is not None:
                         cache_dir = os.path.join(self.cache_dir(), self.NAME)
                     else:
-                        cache_dir = tempfile.mkdtemp(prefix="ast2vec-")
+                        cache_dir = tempfile.mkdtemp(prefix="ml-")
```
Comment (vmarkovtsev) [inline modelforge/model.py]: should be `modelforge`

File: modelforge/model.py (hunk 5 above)
Comment (zurk) [reply modelforge/model.py]: oops, thank you!

**Pull request merged**